### PR TITLE
Support `gql` tags in `@data-eden/codegen`

### PR DESCRIPTION
**Pull request by cafreeman**
## Summary
This is a follow-up to https://github.com/data-eden/data-eden/pull/96 that adds codegen support `gql` tags in addition to `graphql` files.

This PR updates the actual codegen CLI so that it can handle graphql files and embedded graphql tags at the same time. In both cases, the generated code will be written out to a `__generated` directory in the same location as the file being processed (e.g. types for `components/foo.tsx` would end up in `components/__generated/foo.graphql.ts`).

This PR also includes a babel plugin that should be used whenever the consuming app is working with embedded graphql tags. For example usage, see https://github.com/data-eden/data-eden/blob/1138b1e08b671242d646c676ee8d5ed380916ff8/internal-packages/react-graphql-test-app/vite.config.ts#L11

## Follow-up work

- We need to address @rwjblue's comment (https://github.com/data-eden/data-eden/pull/97#discussion_r1190290125) by doing something similar to https://github.com/gajus/babel-plugin-graphql-tag/blob/master/src/index.js#L182-L205

- We will need to expand the functionality here to remove graphql files and generated files after the actual build has taken place (in production). It's important to note that the Babel plugin *relies on the codegen output*, so codegen needs to run prior to a build, even in CI.

However, for production builds, we'd want to go back and remove all those files before bundling.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
 
     strategy:
       matrix:
-        node-version: [14.x, 16.x]
```
Comment (rwjblue) [inline .github/workflows/ci.yml:28]: Let's do the CI work in a separate PR

File: internal-packages/graphql-test-server/src/index.ts [hunk 2]
```diff
@@ -18,6 +18,7 @@ const yoga = createYoga({
       allowArbitraryOperations: true,
       getPersistedOperation(hash: string) {
         console.log('resolving operation for ', hash);
+        console.log(store[hash]);
```
Comment (rwjblue) [inline internal-packages/graphql-test-server/src/index.ts:21]: ✂️

File: internal-packages/react-graphql-test-app/src/components/AddPet.tsx [hunk 3]
```diff
@@ -1,11 +1,62 @@
-import { useMutation } from '@data-eden/react';
+import { graphql, useMutation } from '@data-eden/react';
```
Comment (rwjblue) [inline internal-packages/react-graphql-test-app/src/components/AddPet.tsx:1]: It seems vaguely odd to import this from `/react` package, it isn't really related to react at all is it?

File: internal-packages/react-graphql-test-app/src/components/DisplayPerson.tsx [hunk 4]
```diff
@@ -1,7 +1,11 @@
-import { useQuery } from '@data-eden/react';
+import { useQuery, graphql } from '@data-eden/react';
+import {
+  type PersonQuery,
+  type PersonFieldsFragment,
+  type PersonQueryVariables,
+} from './__generated/DisplayPerson.graphql.js';
```
Comment (rwjblue) [inline internal-packages/react-graphql-test-app/src/components/DisplayPerson.tsx:6]: This seems like a step backwards. Why go back to a top level `__generated` dir? It seems better to emit a sibling file to this one named `DisplayPerson.graphql.d.ts`...

File: packages/codegen/src/babel-plugin.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+  return {
+    name: 'data-eden-codegen',
+    visitor: {
+      Program(path) {
+        program = path;
+      },
+      TaggedTemplateExpression(path, state) {
+        const { node } = path;
+
+        if (!(t.isIdentifier(node.tag) && node.tag.name === options.tagName)) {
```
Comment (rwjblue) [inline packages/codegen/src/babel-plugin.ts:36]: This is not good (reserving a whole variable name). We should be matching an import and allow aliasing. See https://github.com/emberjs/babel-plugin-ember-template-compilation/blob/main/src/plugin.ts#L143 for an example of what we do for `hbs` in Ember land.

**Review by rwjblue — CHANGES_REQUESTED** [review]

File: internal-packages/react-graphql-test-app/src/components/AddPet.tsx (hunk 3 above)
Comment (cafreeman) [reply internal-packages/react-graphql-test-app/src/components/AddPet.tsx:1]: It's just a re-export from `athena`, with the idea being that a react app developer should be install `@data-eden/react` and get everything from there.

File: internal-packages/react-graphql-test-app/src/components/DisplayPerson.tsx (hunk 4 above)
Comment (cafreeman) [reply internal-packages/react-graphql-test-app/src/components/DisplayPerson.tsx:6]: That said, where the actual generated files go is something we can definitely tweak going forward, but I'm gonna leave it as is for now.

File: internal-packages/react-graphql-test-app/src/components/AddPet.tsx (hunk 3 above)
Comment (gabrielcsapo) [reply internal-packages/react-graphql-test-app/src/components/AddPet.tsx:1]: that is a good piece of context. if you are using react just import the react package.

**Review by gabrielcsapo — APPROVED** [review]
based on @rwjblue I agree. Can't wait to see this get merged 🙌

File: packages/codegen/src/babel-plugin.ts (hunk 5 above)
Comment (gabrielcsapo) [reply packages/codegen/src/babel-plugin.ts:36]: https://github.com/data-eden/data-eden/issues/99

**Pull request merged**